docs/examples/plot_tfidfvectorizer.py

```diff
@@ -0,0 +1,190 @@
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.metrics import classification_report
+from sklearn.linear_model import LogisticRegression
+
+
+# limit the list of categories to make running this example faster.
+categories = ['alt.atheism', 'talk.religion.misc']
+train = fetch_20newsgroups(random_state=1,
+                           subset='train',
+                           categories=categories,
+                           )
+test = fetch_20newsgroups(random_state=1,
+                          subset='test',
+                          categories=categories,
+                          )
+                          
+##############################
+# The first transform extract two fields from the data.
+# We take it out form the pipeline and assume
+# the data is defined by two text columns.
+
+class SubjectBodyExtractor(BaseEstimator, TransformerMixin):
+    """Extract the subject & body from a usenet post in a single pass.
+    Takes a sequence of strings and produces a dict of sequences. Keys are
+    `subject` and `body`.
+    """
+    def fit(self, x, y=None):
+        return self
+
+    def transform(self, posts):
+        # construct object dtype array with two columns
+        # first column = 'subject' and second column = 'body'
+        features = np.empty(shape=(len(posts), 2), dtype=object)
+        for i, text in enumerate(posts):
+            headers, _, bod = text.partition('\n\n')
+            bod = strip_newsgroup_footer(bod)
+            bod = strip_newsgroup_quoting(bod)
+            features[i, 1] = bod
+
+            prefix = 'Subject:'
+            sub = ''
+            for line in headers.split('\n'):
+                if line.startswith(prefix):
+                    sub = line[len(prefix):]
+                    break
+            features[i, 0] = sub
+
+        return features
+        
+train_data = SubjectBodyExtractor().fit_transform(train.data)
+test_data = SubjectBodyExtractor().fit_transform(test.data)
+
+######################################
+# The pipeline is almost the same except
+# we remove the custom features.
+
+pipeline = Pipeline([
+    ('union', ColumnTransformer(
+        [
+            ('subject', TfidfVectorizer(min_df=50), 0),
+
+            ('body_bow', Pipeline([
+                ('tfidf', TfidfVectorizer()),
+                ('best', TruncatedSVD(n_components=50)),
+            ]), 1),
+
+            # Removed from the original example as it requires a custom converter.
+            # ('body_stats', Pipeline([
+            #     ('stats', TextStats()),  # returns a list of dicts
+            #     ('vect', DictVectorizer()),  # list of dicts -> feature matrix
+            # ]), 1),
+        ],
+
+        transformer_weights={
+            'subject': 0.8,
+            'body_bow': 0.5,
+            # 'body_stats': 1.0,
```

Comment:
Why do you have this commented out here? If you don't need it, may be remove the line.

Reply:
I hesitated to keep it. It comes from the original examples. I think I wrote the example was more simple than the one from the documentation because it is not yet easy to convert such custom transformer but I let it to let the user know that's missing from the original example.